### ISSUE TITLE
Many settings interfaces are exposed only from internals

### ISSUE THREAD

**Issue by lancegliser** [amcharts/amcharts5#227]
@sassomedia rightly pointed out that some of my implementation code has started to depend on exports from the `@amcharts/amcharts5/.internal/` section in  #219. With thanks to him, I've updated what I could use to more public packages. However there are a number of things I'd like access to but can only get from `.internals` in just the ForceDirected, XYChart, and Sankey I've worked on this week. Could we talk about exposing these outside `.internals` please?

# General

* import { ILegendSettings } from "@amcharts/amcharts5/.internal/core/render/Legend";
* import { Theme as AMChartsTheme } from "@amcharts/amcharts5/.internal/core/Theme";

# XYChart

* import { AxisRenderer } from "@amcharts/amcharts5/.internal/charts/xy/axes/AxisRenderer";
* import { IColumnSeriesSettings } from "@amcharts/amcharts5/.internal/charts/xy/series/ColumnSeries";
* import { IXYChartSettings } from "@amcharts/amcharts5/.internal/charts/xy/XYChart";
* import { IXYCursorSettings } from "@amcharts/amcharts5/.internal/charts/xy/XYCursor";

# ForceDirected

* import { IForceDirectedSettings } from "@amcharts/amcharts5/.internal/charts/hierarchy/ForceDirected";

# Sankey
* import { ISankeySettings } from "@amcharts/amcharts5/.internal/charts/flow/Sankey";

I'm sure once I look at other charts, I'd find similar. The `I*Settings` types seem to be the primary culprit. 

`AxisRenderer` was a bit of a surprise. Reasoning for needing it was this:
```ts
  const [yAxis, setYAxis] = useState<CategoryAxis<AxisRenderer>>();
  // ... Inside a later useEffect
    const xAxis = chart.xAxes.push(
      ValueAxis.new(root, {
        renderer: AxisRendererX.new(root, {}),
      })
    );
    setYAxis(yAxis);
```

I did try to more strongly type `yAxis` as `AxisRendererX`, but the return value of `AxisRendererX.new` is not `AxisRendererX`, it's just `AxisRenderer`. I could _maybe_ do some fancy business for something like that using `ReturnType<AxisRendererX.new>`, but it does seem a bit convoluted.

**Comment by Pauan** [comment]
> I did try to more strongly type yAxis as AxisRendererX, but the return value of AxisRendererX.new is not AxisRendererX

I'm not able to reproduce this, `AxisRendererX.new` correctly returns `AxisRendererX` (just like every other class).

As for your request, we will expose all those things publicly, it was just an oversight on our part.

**Comment by martynasma** [comment]
BTW, `Theme` is already exposed in `index.ts`

**Comment by martynasma** [comment]
Fixed in [5.1.1](https://github.com/amcharts/amcharts5/releases/tag/5.1.1).

## [5.1.1] - 2022-01-31

## Added
- New method `remove(key)` on Template's adapaters.
- Each chart-type package now exports its default theme as `DefaultTheme`. E.g. `am5xy.DefaultTheme`, `am5flow.DefaultTheme`, etc.
- TypeScript: Settings interfaces for all objects are now being exposed via bundled re-export files like `index.ts`, `xy.ts`, etc.
- New class `am5.Modal`.
- New `Container` setting: `reverseChildren`. If set to `true`, will start rendering from the last child, effectively flipping the order in how children of the cotainer are laid out. Useful for [right-to-left setups](
https://www.amcharts.com/docs/v5/concepts/locales/right-to-left-support/).

## Changed
- Charts on iOS devices will now automatically use lesser resolution to work around canvas memory restriction introduced in iOS 15 Safari.
- Charts will now use less built-in layers, and thus less `<canvas>` elements by default, reducing memory consumption.
- All low/high-related settings and fields moved from `CandlestickSeries` to `XYSeries`. Sometimes it could be useful to have low/high on other series even if they are not shown visually.

## Fixed
- `XYCursor` was not being hidden when its `positionX` or `positionY` was changed to `undefined` and `alwaysShow` was set to `false`.
- Auto-wrapped right-aligned labels were being rendered in wrong position.
- Auto-wrapped labels were not being trimmed for whitespace on each line.
- `dropFromPrevious` and `riseFromPrevious` states on `ColumnSeries` were not working properly on a column after a gap in data in case previous data item was outside of selection.
- Emoji characters were not being displayed on circular labels.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
